### PR TITLE
Clean up some compatibility layer in Eauto

### PR DESCRIPTION
**Pull request by ppedrot**
Not much more to say. This should be observationally equivalent.

### CODE REVIEW

**Review by Alizter — APPROVED** [review]

File: tactics/eauto.ml [hunk 1]
```diff
@@ -172,106 +161,89 @@ and prev_search_state = (* for info eauto *)
   | Init
   | State of search_state
 
+(*s Tactics handling a list of goals. *)
+
+(* first_goal : goal list sigma -> goal sigma *)
```
Comment (SkySkimmer) [inline tactics/eauto.ml:166]: What's this comment about?

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@Alizter cleanups don't go in 8.15

File: tactics/eauto.ml (hunk 1 above)
Comment (ppedrot) [reply tactics/eauto.ml:166]: I think that this is a remnant from an even older state of the code.

**Comment by Alizter** [comment]
merging soon

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**